operator-common/src/main/java/io/strimzi/operator/common/operator/resource/concurrent/AbstractResourceOperator.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.operator.resource.concurrent;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.regex.Pattern;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.api.model.ListOptionsBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.FilterWatchListDeletable;
+import io.fabric8.kubernetes.client.dsl.Listable;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.operator.resource.ResourceDiff;
+
+/**
+ * Abstract resource creation, for a generic resource type {@code R}. This class
+ * applies the template method pattern, first checking whether the resource
+ * exists, and creating it if it does not. It is not an error if the resource
+ * did already exist.
+ *
+ * @param <C> The type of client used to interact with kubernetes.
+ * @param <T> The Kubernetes resource type.
+ * @param <L> The list variant of the Kubernetes resource type.
+ * @param <R> The resource operations.
+ */
+public abstract class AbstractResourceOperator<C extends KubernetesClient,
+        T extends HasMetadata,
+        L extends KubernetesResourceList<T>,
+        R extends Resource<T>> {
+    /**
+     * Default reconciliation timeout
+     */
+    private static final long DEFAULT_TIMEOUT_MS = 300_000;
```

Comment:
Is this hardcoded in the original VErt.x operators as well? This should be ideally configured through the operation timeout configured at the operator level.

Reply:
Ok, If it was hardcoded before and nobody complained, I would just leave it that way I guess 🤷. Or if we decide to address it it can definitely be a separate PR in such a case.